### DISCUSSION TITLE
blog/2022/02/spring-boot-live-reload/

### DISCUSSION THREAD

**Discussion by giscus** [d0lim/blog-comments#10] [Announcements]
# blog/2022/02/spring-boot-live-reload/

Spring Boot에서 저장할 때 Live Reload하기

https://d0lim.com/blog/2022/02/spring-boot-live-reload/

**Comment by lkoosss** [comment]
작성해주신 포스팅 도움을 받아서 저는 거의 즉각적으로 reload 되고 있습니다.

저도 추가적으로 내용을 공유드립니다
1. Reload 딜레이 줄이는 방법
  - application.yml 이나 application.properties에 라인추가
      spring.devtools.restart.poll-interval: 50    # Classpath 변경에 대한 Polling 주기시간
      spring.devtools.restart.quiet-period: 10   # 재시작이 트리거 되기 전에 클래스 경로를 변경하지 않는 시간

2. html, css, js의 변경사항도 reload 하도록 Path 추가
  - application.yml 이나 application.properties에 라인추가
      spring.devtools.restart.additional-paths: src/main/webapp  # Classpath 이외에 변경을 감지할 디렉토리 
                                                                                                         (root 프로젝트 아래 경로부터 시작)

  ## 저의 프로젝트 구조  (JSP사용중)
     /src/main
                  /java
                  /resources
                  /webapp
                            /js
                                   /test.js
                            /css
                                   /test.css
                            /WEB-INF
                                   /views
                                        /test.html

3. java 파일 수정시 2번 재시작 되는 것은 spring.devtools.restart.quiet-period 이 너무 짧아서라고 합니다.
   정확한 이유는 모르지만 1500초 정도로 설정하면 증상이 없어지긴 합니다. 근데 기다리는게 더 느려서 저
  는 그냥 2번 재시작되게 냅두고 있어요
   링크) https://github.com/spring-projects/spring-boot/issues/25269

4. devtools 설정이 제대로 되어 있으면 intelliJ의 on frame deactivation / update resources 는 설정하지 않아도 잘 반영됩니다.

**Reply by d0lim** [reply]
안녕하세요, lkoosss님!
추가적인 내용 댓글로 보충해주셔서 감사합니다 🥹
말씀해주신대로 적용한 후 poll-interval과 quiet-period를 이것저것 조절하고 있는데, 이게 또 다시 IntelliJ에 설정한 Save Actions와 꼬일 때가 있네요. 지금으로써는 2번 재시작되게 두는게 편한 듯 하네요. 
이 글이 작성된 지 어언 1년이 지나가던 차라, 저도 약긴 기억에서 잊고 있었는데, Remind 해주셔서 감사합니다. 4번의 내용은 조만간 반영해서 게시글 수정해두도록 하겠습니다.
